src/components/button/toggling-button.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Create Toggling Button
+ * Adds an event listener to toggle the `aria-pressed` value
+ *
+ * @param togglingButton - The button to apply the toggling functionality to
+ */
+export const initTogglingButton = (togglingButton: HTMLButtonElement) => {
+  // Handler for when a toggling button is clicked
+  const onTogglingButtonClick = () => {
+    // Get the current aria-pressed state and toggle it
+    const isAriaPressed = !(
+      togglingButton.getAttribute('aria-pressed') === 'true'
+    );
+    // Update the UI state
+    togglingButton.setAttribute('aria-pressed', String(isAriaPressed));
+  };
+
+  // Clean up event listeners
+  const destroy = () => {
+    togglingButton.removeEventListener('click', onTogglingButtonClick);
+  };
+
+  // Intialize
+  const init = () => {
+    togglingButton.addEventListener('click', onTogglingButtonClick);
+  };
+
+  init();
```

Comment:
I don't think we need a function here since it is single-use. We could inline the addEventListener call. The comment is enough to help understand what is going on

Reply:
Done! https://github.com/cloudfour/cloudfour.com-patterns/pull/1318/commits/b0c54dc7e78a8acb220710ffb7c9079876f3689b